subsys/bluetooth/host/iso.c

```diff
@@ -743,6 +775,13 @@ int bt_iso_chan_send(struct bt_iso_chan *chan, struct net_buf *buf,
 		return -EINVAL;
 	}
 
+	max_data_len = iso_chan_max_data_len(chan, ts);
+	if (buf->len > max_data_len) {
+		BT_DBG("Cannot send %u octets, maximum %u",
+		       buf->len, max_data_len);
+		return -EINVAL;
```

Comment:
Is `EINVAL` really the most suitable error here? Why not e.g. `EMSGSIZE`?

Reply:
Agreed, `EMSGSIZE` is better here.